controllers.py

```diff
@@ -20,20 +20,19 @@ def inserir_diretor():
 def alterar_diretor(id):
     diretor = diretores_from_web(**request.json)
     if id_valida_diretor(id):
-        alter_diretor(**diretor)
+        alter_diretor(id ,**diretor)
         return jsonify(diretores_from_bd(diretor))
     else:
         return jsonify({"Erro":"Diretor não alterado..."})
 
 @app.route("/diretores/<int:id>", methods=['DELETE'])
 def apagar_diretor(id):
-    id_diretor = diretores_from_web(**request.json)
-    try:
-        if id_diretor(id):
-            alter_diretor(**id_diretor)
-            return jsonify(id_diretor_from_bd(id_diretor))
-    except:
-        return({"Erro":"O diretor selecionado não pode ser apagado!"})
+   id_diretor = id_diretor_from_web(**request.json)
+   try:
+       if id_valida_genero(id):
+           delete_dir(**id_diretor)
+           return jsonify(id_genero_from_db(id_diretor))
```

Comment:
você pode fazer assim:

```suggestion
           return None, 204
```

porque o delete não retorna nada.